WalletWasabi/WabiSabi/Client/RoundStatusUpdater.cs

```diff
@@ -0,0 +1,96 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class RoundStatusUpdater : PeriodicRunner
+	{
+		private IWabiSabiApiRequestHandler ArenaRequestHandler { get; }
+		private ImmutableDictionary<uint256, RoundState> RoundStates { get; set; } = ImmutableDictionary.Create<uint256, RoundState>();
+
+		private Dictionary<uint256, List<(TaskCompletionSource<RoundState> Task, Predicate<RoundState> Predicate)>> Awaiters { get; set; } = new();
+
+		public RoundStatusUpdater(TimeSpan requestInterval, IWabiSabiApiRequestHandler arenaRequestHandler) : base(requestInterval)
+		{
+			ArenaRequestHandler = arenaRequestHandler;
+		}
+
+		protected override async Task ActionAsync(CancellationToken cancellationToken)
+		{
+			var responseRoundStates = (await ArenaRequestHandler.GetStatusAsync(cancellationToken).ConfigureAwait(false))
+				.ToDictionary(round => round.Id);
```

Comment:
This is okay but it triggers episodes of neurosis in me.

```suggestion
			var responseRoundStatus = await ArenaRequestHandler.GetStatusAsync(cancellationToken).ConfigureAwait(false);
			var responseRoundStates = responseRoundStatus.ToDictionary(round => round.Id);
```

Reply:
😄